endpoints/openrtb2/auction_test.go

```diff
@@ -2076,6 +2077,245 @@ func TestValidateBidders(t *testing.T) {
+			givenRequest: &openrtb.BidRequest{
+				App:    &openrtb.App{},
+				Device: &openrtb.Device{OS: "iOS", OSV: "14.2", Ext: json.RawMessage(`{"atts":3}`), Lmt: &lmtOne},
+			},
+			expectedLMT: &lmtZero,
+		},
+		{
+			description: "iOS 14.2+ - Status - Not Determined",
+			givenRequest: &openrtb.BidRequest{
+				App:    &openrtb.App{},
+				Device: &openrtb.Device{OS: "iOS", OSV: "14.3", Ext: json.RawMessage(`{"atts":0}`), Lmt: nil},
+			},
+			expectedLMT: &lmtZero,
+		},
+		{
+			description: "iOS 14.2+ - Status - Restricted",
+			givenRequest: &openrtb.BidRequest{
+				App:    &openrtb.App{},
+				Device: &openrtb.Device{OS: "iOS", OSV: "14.3", Ext: json.RawMessage(`{"atts":1}`), Lmt: nil},
+			},
+			expectedLMT: &lmtOne,
+		},
+		{
+			description: "iOS 14.2+ - Status - Denied",
+			givenRequest: &openrtb.BidRequest{
+				App:    &openrtb.App{},
+				Device: &openrtb.Device{OS: "iOS", OSV: "14.3", Ext: json.RawMessage(`{"atts":2}`), Lmt: nil},
+			},
+			expectedLMT: &lmtOne,
+		},
+		{
+			description: "iOS 14.2+ - Status - Authorized",
+			givenRequest: &openrtb.BidRequest{
+				App:    &openrtb.App{},
+				Device: &openrtb.Device{OS: "iOS", OSV: "14.3", Ext: json.RawMessage(`{"atts":3}`), Lmt: nil},
+			},
+			expectedLMT: &lmtZero,
+		},
+		{
+			description: "iOS 14.2+ - Overwrites Existing LMT Value",
+			givenRequest: &openrtb.BidRequest{
+				App:    &openrtb.App{},
+				Device: &openrtb.Device{OS: "iOS", OSV: "14.3", Ext: json.RawMessage(`{"atts":3}`), Lmt: &lmtOne},
+			},
+			expectedLMT: &lmtZero,
+		},
+		{
+			description: "iOS Case Insensitive",
+			givenRequest: &openrtb.BidRequest{
+				App:    &openrtb.App{},
+				Device: &openrtb.Device{OS: "ios", OSV: "14.0", IFA: "", Lmt: nil},
+			},
+			expectedLMT: &lmtOne,
+		},
+		{
+			description: "Unchanged - iOS Version Empty",
+			givenRequest: &openrtb.BidRequest{
+				App:    &openrtb.App{},
+				Device: &openrtb.Device{OS: "iOS", Lmt: nil},
+			},
+			expectedLMT: nil,
+		},
+		{
+			description: "Unchanged - iOS Version Invalid",
+			givenRequest: &openrtb.BidRequest{
+				App:    &openrtb.App{},
+				Device: &openrtb.Device{OS: "iOS", OSV: "14.3", Lmt: nil},
+			},
+			expectedLMT: nil,
+		},
+		{
+			description: "Unchanged - Missing App",
+			givenRequest: &openrtb.BidRequest{
+				App:    nil,
+				Device: &openrtb.Device{OS: "iOS", OSV: "14.2", Ext: json.RawMessage(`{"atts":3}`), Lmt: nil},
+			},
+			expectedLMT: nil,
+		},
+		{
+			description: "Unchanged - Not iOS",
+			givenRequest: &openrtb.BidRequest{
+				App:    &openrtb.App{},
+				Device: &openrtb.Device{OS: "not-ios", OSV: "14.2", Ext: json.RawMessage(`{"atts":3}`), Lmt: nil},
+			},
+			expectedLMT: nil,
+		},
+	}
+
+	for _, test := range testCases {
+		workaroundIOS14PrivacyBug(test.givenRequest)
+		assert.Equal(t, test.expectedLMT, test.givenRequest.Device.Lmt, test.description)
+	}
+}
+
+func TestWorkaroundIOS14PrivacyBugEndToEnd(t *testing.T) {
```

Comment:
This is testing that the workaround logic is being hit during an end-to-end test.

I think the combinatorial details covered by the `TestWorkaroundIOS14PrivacyBug` are sufficient to prove the logic works and this test proves the logic is hit.